SchoolMealBoard/src/main/java/com/zerobase/schoolmealboard/service/impl/MealServiceImpl.java

```diff
@@ -53,75 +53,97 @@ public void fetchAndSaveMealInfo() {
+
+            // API 호출 실패시 로그기록 후 다음으로 넘어감
+            if (response.getStatusCode() != HttpStatus.OK) {
+              log.error("API로부터 데이터를 받아오는 데에 실패했습니다. 상태 코드: {}", response.getStatusCode());
+              continue;
+            }
 
-          // JSON에서 Response로 변환
-          MealResponse mealResponse = objectMapper.readValue(response.getBody(), MealResponse.class);
-          List<MealResponse.MealServiceDietInfo> mealInfoList = mealResponse.getMealServiceDietInfo();
+            // JSON에서 Response로 변환
+            MealResponse mealResponse = objectMapper.readValue(response.getBody(), MealResponse.class);
+            List<MealResponse.MealServiceDietInfo> mealInfoList = mealResponse.getMealServiceDietInfo();
 
-          if (mealInfoList == null) {
-            continue; // 급식 정보가 없으면 다음 날짜로 넘어감
-          }
+            if (mealInfoList == null) {
+              continue; // 급식 정보가 없으면 다음 날짜로 넘어감
+            }
 
-          List<MealResponse.MealRow> rows = mealInfoList.get(1).getRow();  // 데이터 추출
-
-          // 데이터를 순회하며 메뉴와 날짜 DB에 저장
-          for (MealResponse.MealRow row : rows) {
-            String mealNames = row.getMealName();
-            LocalDate mealDate = LocalDate.parse(row.getMealDate(), DATE_FORMAT);
-
-            // 기존 데이터 조회
-            Optional<Meal> existingMealOptional = mealRepository.findBySchoolCodeAndMealDate(school, mealDate);
-
-            if (existingMealOptional.isPresent()) {
-              // 기존 데이터가 있으면 업데이트
-              Meal existingMeal = existingMealOptional.get();
-              existingMeal.setMealNames(mealNames);
-              mealRepository.save(existingMeal);
-            } else {
-              // 기존 데이터가 없으면 새로 추가
-              Meal meal = new Meal();
-              meal.setSchoolCode(school);
-              meal.setMealNames(mealNames);
-              meal.setMealDate(mealDate);
-              mealRepository.save(meal);
+            List<MealResponse.MealRow> rows = mealInfoList.get(1).getRow();  // 데이터 추출
+
+            // 데이터를 순회하며 메뉴와 날짜 DB에 저장
+            for (MealResponse.MealRow row : rows) {
+              String mealNames = row.getMealName();
+              LocalDate mealDate = LocalDate.parse(row.getMealDate(), DATE_FORMAT);
+
+              // 기존 데이터 조회
+              Optional<Meal> existingMealOptional = mealRepository.findBySchoolCodeAndMealDate(
+                  school, mealDate);
+
+              if (existingMealOptional.isPresent()) {
+                // 기존 데이터가 있으면 업데이트
+                Meal existingMeal = existingMealOptional.get();
+                existingMeal.setMealNames(mealNames);
+                mealRepository.save(existingMeal);
+              } else {
+                // 기존 데이터가 없으면 새로 추가
+                Meal meal = new Meal();
+                meal.setSchoolCode(school);
+                meal.setMealNames(mealNames);
+                meal.setMealDate(mealDate);
+                mealRepository.save(meal);
+              }
```

Comment:
이렇게 해보셔도 좋을 것 같아요. Optional type을 변수로 선언하는 것은 안티패턴이라 피하시면 좋을 것 같습니다
```suggestion
              mealRepository.findBySchoolCodeAndMealDate(school, mealDate).ifPresent(existMeal -> 
                  { 
                       existMeal.setMealNames(mealNames);
                       mealRepository.save(existMeal);
                   }).orElseGet(() -> {
                      // 기존 데이터가 없으면 새로 추가
                      Meal meal = new Meal();
                      meal.setSchoolCode(school);
                      meal.setMealNames(mealNames);
                      meal.setMealDate(mealDate);
                      mealRepository.save(meal);
                    });
```

Reply:
@mingming-mentor 
멘토님께서 의도하셨던 건 .map().orElseGet() 이었겠지요..?
ifPresentOrElse()를 활용하여 수정하였는데 괜찮을까요?  
Optional이 편한 줄 알았는데 꽤나 금지하는 것들이 많네요. 덕분에 여러모로 알아갑니다!